### PR TITLE
[PIR] Enable program translator in dy2st mode

### PR DESCRIPTION
**Pull request by kangguangli**
### PR types

Others
### PR changes

Others
### Description

Enable program translator in dy2st mode, but disable in following cases:
1.  program is loaded and run in dy2st mode
2. cinn is enabled
3. composite mechanism is enabled
4. in some unit tests, this is temporarily, we hope them fixed before 12/10

#### Additional Modifications
1. Add attribute `in_pir_pt_mode` for `run_porgram` op
2. Introduce flag `enable_pir_with_pt_in_dy2st` 

Both of them is used to control whether Pir is enabled in dy2st mode. 

#### Others
Pcard-67164

### CODE REVIEW

**Review by SigureMo — APPROVED** [review]
LGTMeow

File: paddle/fluid/eager/to_static/run_program_op_node.h [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 #pragma once
 
+#include <iostream>
```
Comment (Aurelius84) [inline paddle/fluid/eager/to_static/run_program_op_node.h:17]: 这个应该是调试时忘记删除了吧

File: python/paddle/jit/dy2static/partial_program.py [hunk 2]
```diff
@@ -799,6 +799,18 @@ def _prepare_attributes(self):
...
                 )
             )
+
+        pir_dy2st_flag = 'FLAGS_enable_pir_with_pt_in_dy2st'
+        in_pir_pt_mode = get_flags(pir_dy2st_flag)[pir_dy2st_flag]
+        is_prim_enabled = (
+            core._is_fwd_prim_enabled() or core._is_bwd_prim_enabled()
+        )
+        in_cinn_backend = self._backend == "CINN"
+        is_cinn_enabled = self._build_strategy.build_cinn_pass
```
Comment (Aurelius84) [inline python/paddle/jit/dy2static/partial_program.py:809]:
```suggestion
        is_cinn_enabled = self._build_strategy.build_cinn_pass or self._backend == "CINN"
```
可以省去一个变量

**Review by Aurelius84 — APPROVED** [review]
可以考虑提个commit fix下或者合入后单独提PR

**Comment by kangguangli** [comment]
> Details

好的，下个PR修复。

**Review by heavengate — APPROVED** [review]

**Review by vivienfanghuagood — APPROVED** [review]

**Pull request merged**